src/parquet/column/reader.h

```diff
@@ -95,6 +100,11 @@ class ColumnReader {
   // The number of values from the current data page that have been decoded
   // into memory
   int num_decoded_values_;
+
+  // We need to store the data page for BYTE_ARRAY and FLBA since the corresponding types
+  // store pointers into these pages
+  // The user is responsible for freeing these DataPages by calling ClearDataPages()
+  std::vector<std::shared_ptr<Page> > byte_pages_;
```

Comment:
I did not look into the lifetime of these DataPages carefully. The above vector is actually redundant.